.github/workflows/build.yml

```diff
@@ -109,41 +76,33 @@ jobs:
         uses: ocaml/setup-ocaml@v3
         with:
           ocaml-compiler: ${{ matrix.ocaml-compiler }}
-          dune-cache: true
-          opam-pin: false
-
-      - run: opam install conf-pkg-config
-        if: runner.os == 'Windows'
 
       - run: opam install . --best-effort
-        if: ${{ matrix.skip-test }}
-
-      - run: opam install . --with-test
-        if: ${{ !matrix.skip-test }}
+        if: matrix.ocaml-compiler < 4.14
 
-      - run: opam exec -- make all
-        if: ${{ !matrix.skip-test }}
+      - run: opam install . --deps-only --with-test --with-doc
+        if: matrix.ocaml-compiler >= 4.14
 
-      - run: opam exec -- make tests
-        if: ${{ !matrix.skip-test }}
+      - run: opam exec -- dune build @all @runtest
+        if: matrix.ocaml-compiler >= 4.14 && matrix.ocaml-compiler < 5.0
 
       - run: opam exec -- dune build @all @runtest --profile using-effects
-        if: ${{ !matrix.skip-effects }}
+        if: matrix.ocaml-compiler >= 5.0
 
       - run: opam exec -- git diff --exit-code
-        if: ${{ !matrix.skip-test }}
+        if: matrix.ocaml-compiler >= 4.14
 
-      - name: build doc
-        if: ${{ !matrix.skip-doc && github.event_name == 'push' && github.ref_name == 'master'}}
+      - name: Build doc
+        if: matrix.os == 'ubuntu-latest' && matrix.ocaml-compiler == '5.2' && github.event_name == 'push' && github.ref_name == 'master'
```

Comment:
I would prefer to only bump version once when it come to change the version we generate the doc with.